content/docs/tutorial/dynamic-enum.mdx

```diff
@@ -0,0 +1,259 @@
+
+```js
+import { Unwrapped, WithOptionLabel } from '@jsonforms/material-renderers';
+
+const { MaterialEnumControl } = Unwrapped;
+
+...
+
+<MaterialEnumControl
+      {...props}
+      options = {options}
+      handleChange = {handleChange}
+    />
+...
+```
+
+With the `MaterialEnumControl`in place the main question remains how to set the `options` and the `handleChange` attribute.
+To determine the available options, we need to access the API.
+And to implement the `handleChange` function, we need access to the `dependent` field in the schema.
+
+#### Accessing Schema Data
+
+The `endpoint` and `dependent` fields can be obtained from the schema object provided to the custom renderer via JSON Forms.
+Since these fields are not part of the standard JSON schema type in JSON Forms, we must add them to the schema's interface and access them as follows:
+
+```js
+type JsonSchemaWithDependenciesAndEndpoint = JsonSchema & {
+  dependent: string[];
+  endpoint: string;
+};
+export const Country = (
+  props: ControlProps & OwnPropsOfEnum & WithOptionLabel & TranslateProps
+) => {
+...
+
+  const schema = props.schema as JsonSchemaWithDependenciesAndEndpoint;
+  const endpoint = schema.endpoint;
+  const dependent = schema.dependent
+...
+}
+```
+
+#### Country Renderer Implementation
+
+The country renderer uses the `APIContext` to query the API and fetch the available options.
+We utilize the `useEffect` hook to initialize the options.
+While waiting for the API response, we set the available options to empty and display a loading spinner.
+In the `handleChange` function, we set the new selected value and reset all dependent fields;
+When changing the country, both the region and city will be reset to `undefined`.
+
+```js
+import { Unwrapped, WithOptionLabel } from '@jsonforms/material-renderers';
+
+const { MaterialEnumControl } = Unwrapped;
+
+type JsonSchemaWithDependenciesAndEndpoint = JsonSchema & {
+  dependent: string[];
+  endpoint: string;
+};
+
+export const Country = (
+  props: ControlProps & OwnPropsOfEnum & WithOptionLabel & TranslateProps
+) => {
+  const { handleChange } = props;
+  const [options, setOptions] = useState<string[]>([]);
+  const api = React.useContext(APIContext);
+  const schema = props.schema as JsonSchemaWithDependenciesAndEndpoint;
+
+  const endpoint = schema.endpoint;
+  const dependent: string[] = schema.dependent ? schema.dependent : [];
+
+  useEffect(() => {
+    api.get(endpoint).then((result) => {
+      setOptions(result);
+    });
+  }, []);
+
+  if (options.length === 0) {
+    return <CircularProgress />;
+  }
+
+  return (
+    <MaterialEnumControl
+      {...props}
+      handleChange={(path: string, value: any) => {
+        handleChange(path, value);
+        dependent.forEach((path) => {
+          handleChange(path, undefined);
+        });
+      }}
+      options={options.map((option) => {
+        return { label: option, value: option };
+      })}
+    />
+  );
+};
+```
```

Comment:
Please add the wrapping code for the renderers, i.e. `withJsonFormsControlProps` etc. here and in the other listings.